### PR TITLE
embed: add titles (and description) to mobile viewports

### PR DESCRIPTION
**Pull request by matallo**
re: https://github.com/CartoDB/cartodb/issues/13498

added titles to the mobile version of the embeds:

- when the map has legends the titles will be embedded in the legends #meta
- if the map has no legends the title will be added as an overlay that can be toggled to show/hide the description

---

**acceptance**

titles and description should appear in the different viewports for the next cases with layers and without layers

viewports:

- `> 1200`
  without legends + with description / without description
  pageshot of untitled map 1 2018-02-02-1947 06

  with legends + with description / without description
  pageshot of untitled map 2018-02-02-1947 11

- `> 600 < 1200`
  without legends + with description (expanded)
  pageshot of untitled map 1 import 1 untitled map 1 import 1 2018-02-02-1945 47

  without legends + with description (collapsed)
  pageshot of untitled map 1 import 1 untitled map 1 import 1 2018-02-02-1945 51

  without legends + without description
  pageshot of untitled map 1 2018-02-02-1945 25

  with legends + with description
  pageshot of untitled map 1 import 1 untitled map 1 import 1 2018-02-02-1946 00

  with legends + without description
  pageshot of untitled map 2018-02-02-1945 36

- `< 600`
  without legends + with description
  pageshot of untitled map 1 import 1 untitled map 1 import 1 2018-02-02-1943 43

  without legends + without description
  pageshot of untitled map 1 2018-02-02-1943 14

  with legends + with description (map tab)
  pageshot of untitled map 1 import 1 untitled map 1 import 1 2018-02-02-1943 53 1

  with legends + with description (legends tab)
  pageshot of untitled map 1 import 1 untitled map 1 import 1 2018-02-02-1944 06

  with legends + without description (map tab)
  pageshot of untitled map 2018-02-02-1943 27

  with legends + without description (legends tab)
   pageshot of untitled map 2018-02-02-1943 33

### CODE REVIEW

**Comment by matallo** [comment]
tests pending

File: lib/assets/javascripts/cartodb3/public_editor.js [hunk 1]
```diff
@@ -29,8 +29,18 @@ var configModel = new ConfigModel({ base_url: baseURL });
...
 
+function stripHTML (input, allowed) {
+  allowed = (((allowed || '') + '').toLowerCase().match(/<[a-z][a-z0-9]*>/g) || []).join('');
+  var tags = /<\/?([a-z][a-z0-9]*)\b[^>]*>/gi;
+  if (!input || (typeof input !== 'string')) return '';
+  return input.replace(tags, function ($0, $1) {
+    return allowed.indexOf('<' + $1.toLowerCase() + '>') > -1 ? $0 : '';
+  });
+}
+
```
Comment (elenatorro) [inline lib/assets/javascripts/cartodb3/public_editor.js]: I would move this function to a separate file and use it as a helper.

Comment (matallo) [reply lib/assets/javascripts/cartodb3/public_editor.js]: actually that already existed as it is in utils 🙈

**Comment by elenatorro** [comment]
LGTM

**Comment by matallo** [comment]
retest this, please

**Comment by elenatorro** [comment]
Acceptance done, LGTM 🎉

**Pull request merged**